experimental/single_number_formatter/src/provider.rs

```diff
@@ -0,0 +1,179 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+// Provider structs must be stable
+#![allow(clippy::exhaustive_structs, clippy::exhaustive_enums)]
+
+//! Data provider struct definitions for this ICU4X component.
+//!
+//! Read more about data providers: [`icu_provider`]
+
+use alloc::borrow::Cow;
+use icu_provider::{yoke, zerofrom};
+use tinystr::TinyAsciiStr;
+use zerovec::{VarZeroVec, ZeroMap};
+
+#[icu_provider::data_struct(CurrencyEssentialV1Maker = "currency/essential@1")]
+#[derive(Default, Clone, PartialEq, Debug)]
+#[cfg_attr(
+    feature = "datagen",
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_singlenumberformatter::provider),
+)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[yoke(prove_covariance_manually)]
+pub struct CurrencyEssentialV1<'data> {
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub currency_patterns_map: ZeroMap<'data, TinyAsciiStr<3>, CurrencyPatterns>,
```

Comment:
use `UnvalidatedTinyAsciiStr`

Reply:
I think we need to enforce Ascii string, because the iso code must be ascii.

Reply:
Keys don't have to be valid, if they're garbage the lookup just won't work (GIGO).

Reply:
the iso code is intended to be ascii.

Reply:
aha. ... okay